packages/google_maps_flutter/google_maps_flutter/ios/Classes/GoogleMapPolylineController.m

```diff
@@ -45,17 +52,42 @@ - (void)setPoints:(NSArray<CLLocation*>*)points {
   }
   _polyline.path = path;
 }
+- (void)setPattern:(NSArray<FLTPolylinePattern*>*)pattern {
+  _pattern = pattern;
+  NSMutableArray* styles = [[NSMutableArray alloc] init];
+  NSMutableArray* lengths = [[NSMutableArray alloc] init];
```

Comment:
These can be pre-created with size equal to `_pattern`'s size to avoid unnecessary reallocations.